### PR TITLE
Improve fatal message when creating the seed factory fails

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Improve fatal message when creating the seed factory fails.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/cmd/kubermatic-controller-manager/main.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ Please install the VerticalPodAutoscaler according to the documentation: https:/
 	rootCtx, rootCancel := context.WithCancel(context.Background())
 	seedGetter, err := provider.SeedGetterFactory(rootCtx, mgr.GetClient(), options.dc, options.dcFile, options.dynamicDatacenters)
 	if err != nil {
-		log.Fatal(err)
```
Comment (alvaroaleman) [inline api/cmd/kubermatic-controller-manager/main.go:102]: Nit: Wouldn't it be nicer if this was a `log.With("error", err).Fatal(...)`?

Comment (mrIncompetent) [reply api/cmd/kubermatic-controller-manager/main.go:102]: Oh great idea, i'll update all fatals here with that

**Pull request merged**